duckduckgo/macos-browser
# Fix URLs with port and replace spaces in query with %20

Task/Issue URL: https://app.asana.com/0/1199237043630360/1200001872934852
Tech Design URL:
CC:

**Description**:
- Fixes localhost:5000 interpretation
- Replaces spaces in query edited manually with %20

**Steps to test this PR**:
1. Validate queries like "localhost:5000" and "something.local:9100" try opening local URLs instead of SERP
2. Search for something; Edit query in the address bar adding spaces to the q= argument value, spaces should be replaces with %20
3. Verify that define:foo triggers a search
4. Verify that Zoom links still work
5. Verify that sms://+44123123123 tries to open Messages.app
6. Verify that mailto:test@example.com tries to open a system default Mail app
7. Verify that test://hello triggers a search, because this scheme is not supported by any app in the system
8. Verify that "localdomain" triggers a search
9. Verify that "localhost" opens localhost



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
